cmd/influxd/upgrade/config.go

```diff
@@ -168,36 +142,70 @@ func (cu *configUpgrader) save(config map[string]interface{}, path string) error
 
 // Credits: @rogpeppe (Roger Peppe)
 
-func (cu *configUpgrader) transform(x map[string]interface{}) map[string]interface{} {
+func (cu *configUpgrader) transform(v1Config map[string]interface{}) map[string]interface{} {
 	res := make(map[string]interface{})
 	for old, new := range cu.rules {
-		val, ok := cu.lookup(x, old)
-		if ok {
-			if transform, ok := cu.valueTransforms[new]; ok {
-				val = transform(val)
-			}
+		if val, ok := cu.lookup(v1Config, old); ok {
 			res[new] = val
 		}
 	}
 
+	// Special case: flip the value for pprof.
+	if val, ok := cu.lookup(v1Config, "http.pprof-enabled"); ok {
+		if b, ok := val.(bool); ok {
+			res["pprof-disabled"] = !b
+		}
+	}
+
+	// Special case: ensure query settings are valid.
+	fixQueryLimits(res)
+
 	return res
 }
 
-func (cu *configUpgrader) lookup(x map[string]interface{}, path string) (interface{}, bool) {
+// fixQueryLimits ensures that all query-related config settings are compatible
+// with the upgraded value of the 'query-concurrency' setting.
+func fixQueryLimits(v2Config map[string]interface{}) {
+	concurrencyVal, ok := v2Config["query-concurrency"]
+	if !ok {
+		return
+	}
+	var concurrency int64
+	switch c := concurrencyVal.(type) {
+	case int:
+		concurrency = int64(c)
+	case int32:
+		concurrency = int64(c)
+	case int64:
+		concurrency = c
+	default:
+		concurrency = 0
+	}
+	if concurrency == 0 {
+		// The upgrade process doesn't generate a value for query-queue-size, so if
+		// query-concurrency is 0 / unset then it's safe to leave query-queue-size unset.
+		return
+	}
+
+	// When query-concurrency is > 0, query-queue-size must also be > 0.
+	v2Config["query-queue-size"] = concurrency
```

Comment:
There's a risk here that:
* Somebody will pick a huge value for query-concurrency
* We write the same value into query-queue-size
* At startup, the server uses the huge value for query-queue-size as the size of a new `chan`, consuming a lot of memory all at once

I'm assuming the risk is low now that we support 0 for unlimited, so most people who would be writing huge concurrency values can now write 0 instead.